src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class Duke {
+    public static ArrayList<Task> tasks = new ArrayList<Task>();
+    public static final Scanner SCANNER = new Scanner(System.in);
+
+    public static void main(String[] args) {
```

Comment:
This method seems to be very long, spanning from line 11 to 119 (108 lines!), try to extract the common portions of code within this method into multiple functions and practice SLAP to reduce the method size. This will improve the code quality and make the code more readable. 

It seems like no error handling has been done with exceptions in this part of the code, perhaps try and use exceptions to handle the errors in a succinct manner and avoid increasing the complexity of the code?

https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard